### DISCUSSION TITLE
help needed - Files skipped for signing

### DISCUSSION THREAD

**Discussion by swaroopar** [jreleaser/jreleaser#1409] [Q&A] [answered]
Hi All, 

need some help. I tried to figure it out myself but no luck. 
Signing is always being skipped in my job. Can you please help me figure out the issue?

[INFO] Calculating checksums for distributions and files
[INFO]   [checksum] No files configured for checksum. Skipping
[INFO] Cataloging artifacts
[INFO]   [sbom] Cataloging is not enabled. Skipping
[INFO] Signing distributions and files
[INFO]   [sign] No files configured for signing. Skipping
[INFO] Deploying Maven artifacts
[INFO]   [maven] Deploying all staged artifacts

job - https://github.com/eclipse-xpanse/xpanse/actions/runs/5228358586/jobs/9440647354
pom - https://github.com/eclipse-xpanse/xpanse/blob/feature/releaseUpload/pom.xml#L109
worfklow - https://github.com/eclipse-xpanse/xpanse/blob/feature/releaseUpload/.github/workflows/release.yml#L69

trace log attached. [trace.log](https://github.com/jreleaser/jreleaser/files/11711606/trace.log)

**Comment by aalmiray** [comment]
File signing is a separate step from artifact deployment, the latter also performs signing.

Your project has no `<distributions>`, `<assemble>`, nor `<files>` configured. These blocks provide inputs to the signing step. Given that it seems you want to deploy artifacts to Maven Central you don't need these blocks, thus the message you're getting in the log is OK. 

OTOH the deploy phase fails because pom files are missing required information. Have a look at the errors reported by pomchecker found in the log.

**Reply by swaroopar** [reply]
@aalmiray thank you for the reply. 
Earlier when I ignored the errors from POM.. the upload failed on Maven Central since the .asc files were missing. That's why i simply enabled the checks again to see if that makes a difference (based on the discussion here #1118 )
we got this error on the Maven Central. How can I have the .asc files generated from jreleaser?


![image](https://github.com/jreleaser/jreleaser/assets/54129659/81c4290a-d197-4ca6-be91-c2437706919f)

**Reply by swaroopar** [reply]
@aalmiray I added the files, but still nothing was signed again. 
And the deploy fails with another error'.
`Error:  Failed to execute goal org.jreleaser:jreleaser-maven-plugin:1.6.0:release (default-cli) on project xpanse-parent: Execution default-cli of goal org.jreleaser:jreleaser-maven-plugin:1.6.0:release failed: Cannot invoke "org.bouncycastle.openpgp.PGPPrivateKey.getKeyID()" because "<parameter2>" is null -> [Help 1]`

Job - https://github.com/eclipse-xpanse/xpanse/actions/runs/5229908918/jobs/9443177179

Can you please help here?

P.S - I have added "verifyPom" as false.

**Reply by aalmiray** [reply]
Please attach a copy of the full stacktrace found at `target/jreleaser/trace.log`

**Reply by swaroopar** [reply]
@aalmiray sorry missed that. trace.log file is now attached.
[trace.log](https://github.com/jreleaser/jreleaser/files/11713196/trace.log)

**Reply by aalmiray** [reply, marked as answer]
Is your intention to publish `*.jar` and `*.pom` files as github release assets? If not then remove the `<files>` section. 

Did you configure a GPG public/private key combination that can be used for signing?

**Reply by swaroopar** [reply]
Ah ok, I misunderstood your first answer. No, we do not need files as github release assets. I will remove the `<files>`. 
Yes, the keys are configured and added as secrets to GitHub. All configured secrets are injected as environment vars:

[DEBUG]   JRELEASER_GITHUB_TOKEN
[DEBUG]   JRELEASER_GPG_PASSPHRASE
[DEBUG]   JRELEASER_GPG_PUBLIC_KEY
[DEBUG]   JRELEASER_GPG_SECRET_KEY
[DEBUG]   JRELEASER_NEXUS2_MAVEN_CENTRAL_PASSWORD
[DEBUG]   JRELEASER_NEXUS2_MAVEN_CENTRAL_USERNAME

latest job (switched to Github actions) https://github.com/eclipse-xpanse/xpanse/actions/runs/5239839787/jobs/9460049635
[trace.log](https://github.com/jreleaser/jreleaser/files/11718550/trace.log)

**Reply by swaroopar** [reply]
@aalmiray It finally worked. I added all GPG related vars to the Java setup step and the upload worked. 
Thank again for your support.
